#170947605: Users should be able to Login

Kagorora opened the pull request:
#### What does this PR do?

- Allow users to Login in the system

#### Description of Task to be completed?

- [x] Added Login fronted design

- [x] Added Login Redux functionality

- [x] Added Error Handling

- [x] Added Input validations

- [x] Test components

#### How should this be manually tested?

- clone the project by running `git clone https://github.com/Stackup-Rwanda/knights-bn-frontend.git`

- open the project `cd knights-bn-frontend` then `code .`

- Install dependencies `npm install`

- Run the project `npm run dev-start`

- click on login on the nav-bar

#### Heroku Link

       https://knights-bn-f-ft-login-1-aujgoc.herokuapp.com/

#### Mockup Link

       https://xd.adobe.com/view/ddb6dde0-8b48-4220-6a0f-493993ea9646-ab87/

#### Any background context you want to provide?

- N/A

#### What are the relevant pivotal tracker stories?

- [170947605](https://www.pivotaltracker.com/story/show/170947605)

#### Screenshots (if appropriate)

<img width="1440" alt="Screen Shot 2020-04-30 at 09 42 14" src="https://user-images.githubusercontent.com/48021034/80685047-e6e2c380-8ac6-11ea-9d8a-2d5da72c1116.png">


#### Questions:

- N/A

Kagorora committed:
ft(login): add Login feature

- add Login functionality

[Finishes #170947605]

Niyonsengaeric committed:
Refactor components

- make components reusable

Niyonsengaeric committed:
170947605-ft(thunk):add re-usable thunk
- make thunk reusable
[Finishes #170947605]

Niyonsengaeric committed:
170947605-ft(thunk):add re-usable thunk

- make thunk reusable

Niyonsengaeric committed:
170947605-ft(thunk):add re-usable thunk
- make thunk reusable
[Finishes #170947605]

170947605-ft(thunk):add re-usable thunk

- make thunk reusable

Refactor components

- make components reusable

Niyonsengaeric committed:
Merge pull request #11 from Stackup-Rwanda/refactor-login-compontns

- Refactor components
- Add validations

Niyonsengaeric committed:
Test components.

- test the button and text components
§

EugeneMunya committed:
170947605-ft(thunk): add reusable thunk
[Finishes #170947605]

EugeneMunya committed:
170947605-ft(thunk): add thunk
- add reusable thunk
[Finishes #170947605]

EugeneMunya committed:
make thunk reusable

EugeneMunya committed:
Merge pull request #16 from Stackup-Rwanda/ft-redux-thunk-170947605

Make thunk reusable

Niyonsengaeric committed:
ft(login): Add login page

- add validations

[Finishes #170947605]

kagabof commented on .circleci/config.yml:
```diff
@@ -21,3 +21,9 @@ jobs:
+  workflow:
+    jobs:
+    - build
+    
```
Add a newline in this file.

kagabof commented on src/components/Alert.js:
```diff
@@ -0,0 +1,12 @@
+import { Alert } from 'react-bootstrap';
+
+const AlertComponent = (props) => {
+  const { variant, alertMsg } = props;
```
You can destructure directly in the function parameters without creating a new variable.

kagabof commented on src/components/Button.js:
```diff
@@ -0,0 +1,18 @@
+import React from 'react';
+import { Button } from 'react-bootstrap';
+
+function SubmitButton(props) {
```
please use S6 function syntax.

kagabof commented on src/components/Nav.js:
```diff
@@ -7,14 +7,16 @@ const Nav = () => {
   return (
```
There is no need of return statement.

kagabof commented on src/redux/actions/actions.js:
```diff
@@ -0,0 +1,51 @@
+  FETCH_USERS_FAILURE,
+  VALIDATE_INPUT,
+  USER_LOGIN_SUCCESS,
+} from './actionTypes';
```
You can use a stare to import all functions in that file. ex: ```import * as type from './actionTypes' ```

kagabof commented on src/redux/reducers/userReducer.js:
```diff
@@ -0,0 +1,40 @@
+  FETCH_USERS_SUCCESS,
+  FETCH_USERS_FAILURE,
+  USER_LOGIN_SUCCESS,
+} from '../actions/actionTypes';
```
You can use a stare to import all functions in that file. ex: ```import * as type from './actionTypes'```

kagabof commented on src/views/Home.js:
```diff
@@ -0,0 +1,9 @@
+import React from 'react';
+
+export default function Home() {
```
please use es6 syntax.

kagabof commented on src/views/Login.js:
```diff
@@ -0,0 +1,133 @@
+      error: " ",
+      loggedIn: false,
+    };
+  }
```
In es6 you don't need a constructor.

kagabof commented on src/views/Login.js:
```diff
@@ -0,0 +1,133 @@
+      </div>
+    );
+  }
+  login = async (e) => {
```
Just put this function before render method.

kagabof commented on src/views/Login.js:
```diff
@@ -0,0 +1,133 @@
+            </p>
+          </span>
+          <p>Login with so social media</p>
+          <div className="Google-login-btn">
```
Please refactor this file by making some of the parts of its components.

Kagorora replied:
@kagabof This will be done when implementing the social login

kagabof commented on src/views/LoginPage.js:
```diff
@@ -0,0 +1,10 @@
+import React from 'react';
+import Login from './Login';
+
+export default function LoginPage() {
```
Please use the es6 syntax.

kagabof reviewed (changes requested):
Please make sure you solved those issues.

Niyonsengaeric committed:
Increase test coverage

- test Login compont
- remove unused validations

Niyonsengaeric committed:
Increase test coverage

- test Login compont

Niyonsengaeric committed:
Merge pull request #17 from Stackup-Rwanda/ft-test-components-#1

Ft test components #1

Kagorora committed:
Merge pull request #18 from Stackup-Rwanda/ft-test-components-#1

Increase test coverage

kagabof commented on __tests__/Footer.test.js:
```diff
@@ -0,0 +1,13 @@
+import '@testing-library/jest-dom';
+import Footer from '../src/components/Footer';
+
+describe('=============> Footer <=============', () => {
```
Please describe properly.

kagabof commented on __tests__/Header.test.js:
```diff
@@ -0,0 +1,17 @@
+import '@testing-library/jest-dom';
+import Header from '../src/components/Header';
+
+describe('===== Test Header =====', () => {
```
Please describe properly.

kagabof commented on __tests__/Login.test.js:
```diff
@@ -0,0 +1,166 @@
+        <View {...props} />
+      </MemoryRouter>,
+    );
+    console.log('<<<<<</////>>>>', wrapper.props());
```
remove those consoles

kagabof commented on __tests__/Login.test.js:
```diff
@@ -0,0 +1,166 @@
+      </MemoryRouter>,
+    );
+    console.log('<<<<<</////>>>>', wrapper.props());
+    console.log('<<<<<', wrapper.find('Button').simulate('click'));
```
remove this console.

kagabof commented on __tests__/Login.test.js:
```diff
@@ -0,0 +1,166 @@
+      </MemoryRouter>,
+    );
+    console.log('<<<<<</////>>>>', wrapper.props());
+    console.log('<<<<<', wrapper.find('Button').simulate('click'));
```
remove those console

kagabof commented on __tests__/__snapshots__/Nav.spec.js.snap:
```diff
@@ -0,0 +1,44 @@
+// Jest Snapshot v1, https://goo.gl/fbAQLP
```
please ignore those snapshots.

Kagorora committed:
ft(login): Login Feature

- setup redux
- add login functionality
- test login using enzyme

[Finishes #170947605]

kagabof commented on Untitled-1:
```diff
@@ -0,0 +1,17 @@
+module.export = {
```
Name property this file.

kagabof commented on src/jest.config.js:
```diff
@@ -0,0 +1,20 @@
+// module.exports = {
```
why do we have commented code?

kagabof reviewed (changes requested):
Why do we have two jest.config.js?
I need you to test you to increase your test coverage, it has to be above 99%.

kagabof reviewed (approved):
Well done.
